### ISSUE TITLE
Dangerously windy ? (Stuck)

### ISSUE THREAD

**Issue by 41south** [poblabs/weewx-belchertown#175]
Hi Pat

For the last 24 hrs or so all of the current conditions seem to say 'Dangerously Windy' along with whatever else there is. I'm not quite sure where to look to see if that wording is stuck somewhere or it's just a Darksky problem? It's certainly not right :)

**Comment by 41south** [comment]
OK, found the darksky_forecast.json file and it's in there in the 'Summary', so probably a Darksky issue. I've deleted it hoping it will download a new one next archive period and will have another look.

**Comment by 41south** [comment]
Definitely a darksky issue - closing :)

**Comment by 41south** [comment]
Pat is there a way to disable this summary for now? Even a short term hack will do

**Comment by biltmorelaker** [comment]
@41south In my weewx.conf under [[Belchertown]] [[[Extras]]] I have:
 ```
# Forecast Settings
            forecast_enabled = 1
            darksky_secret_key = my_secret_key
            darksky_units = us
            forecast_alert_enabled = 1
```
If you set 
`forecast_enabled = 0`
will that shut it off?

**Comment by biltmorelaker** [comment]
Hmm. @41south, I just looked at your website (http://41south.net.nz/). Where is Dangerously Windy showing?

**Comment by poblabs** [comment]
I'm seeing it (screenshot below). To disable that line - without disabling everything else - you can delete this line in index.html.tmpl. Like you said; it's a hack 

https://github.com/poblabs/weewx-belchertown/blob/master/skins/Belchertown/index.html.tmpl#L189


![image](https://user-images.githubusercontent.com/3484775/59568453-e5dcab00-9048-11e9-8169-dc17040b31de.png)

**Comment by biltmorelaker** [comment]
Ah. Sorry. Now I see it. I must have not been paying attention to what I was looking at.

**Comment by 41south** [comment]
Cool, thanks Pat, I'll take the hack for now, not sure where or why that is coming from Darksky. It's been pretty much calm here for the past few days!

**Comment by 41south** [comment]
It's stubborn! I assume you mean the copy in /etc/weewx/skins/Belchertown? I edited that and it's still there 😕 Restarted weewx, reloaded the web page

**Comment by poblabs** [comment]
I'd send them an email from your darksky portal asking them for help with the stuck status. 

Yes, /etc/weewx/skins/Belchertown/index.html.tmpl should do it. Just completely delete that `$current_obs_summary` and that should do it - even without a restart (but a wee_reports)

**Comment by 41south** [comment]
Yeah will send them a message. I deleted that line but it's still there. And I'm certain I'm editing the right file (mind you I haven't had coffee yet this morning 😆 )
![Screen Shot 2019-06-17 at 8 22 38 AM](https://user-images.githubusercontent.com/1725001/59569117-1835e900-90d9-11e9-92b3-63b980170276.png)

**Comment by 41south** [comment]
Ah - it's being overwritten by something out of the live data I think

**Comment by poblabs** [comment]
Oh you're right - JavaScript is probably taking over a little. Rename the `class` in this div to add a 1 at the end.

https://github.com/poblabs/weewx-belchertown/blob/master/skins/Belchertown/index.html.tmpl#L188

That should disable the JavaScript as well

**Comment by 41south** [comment]
Great, that got it. Thanks.

I'll close this and email Darksky

**Comment by 41south** [comment]
Ok sorry but this isn't getting any less confusing. If I run the api from my browser I get this back
`{"latitude":-41.2345,"longitude":174.8765,"timezone":"Pacific/Auckland","currently":{"time":1560717194,"summary":"Clear","icon":"clear-day","precipIntensity":0,"precipProbability":0,"temperature":52.68,"apparentTemperature":52.68,"dewPoint":34.87,"humidity":0.51,"pressure":1011.11,"windSpeed":4.3,"windGust":18.68,"windBearing":270,"cloudCover":0.05,"uvIndex":0,"visibility":7.034,"ozone":325.6},"hourly":{"summary":"Partly cloudy throughout the day.","icon":"rain","data":[{"time":1560715200,"summary":"Clear","icon":"clear-`

No sign of a Dangerously Windy. Does this query work differently to what the skin is doing?

**Comment by poblabs** [comment]
Maybe. Check your html folder/json/darksky_forecast.json and see if it's in
there. I think under the "currently" array section. In nano you can Ctrl+w
to search for it.

**Comment by 41south** [comment]
Yes it's in the json file
![Screen Shot 2019-06-17 at 9 23 06 AM](https://user-images.githubusercontent.com/1725001/59569719-9b5b3d00-90e1-11e9-95fd-4ca07e936632.png)

**Comment by poblabs** [comment]
Check your syslog for any errors on rewriting that file. Or permission
denied or when the last time it was download was.

Search syslog with Ctrl+w for that same file name

**Comment by 41south** [comment]
It's up to date and I don't see any errors and the timestamp in the file is good  @ 1560720621
-rw-r--r-- 1 root root 25818 Jun 17 09:30 darksky_forecast.json

**Comment by poblabs** [comment]
I'm curious and confused.  Delete the file again, let it automatically download a new one on the next archive. 

Then paste the full output when running their api from your browser. They should be matching up 100% on both of those tests.

**Comment by 41south** [comment]
SO the new json has this;
{"latitude":-41.3194,"longitude":174.818,"timezone":"Pacific/Auckland","currently":{"time":1560740120,"summary":"Dangerously Windy","icon":"clear-day","nearestStormDistance":0,"precipIntensity":0.06,"precipProbability":0.12,"precipType":"rain","temperature":12.64,"apparentTemperature":12.64,"dewPoint":2.19,"humidity":0.49,"pressure":1010.96,"windSpeed":5.22,"windGust":21.12,"windBearing":181,"cloudCover":0.04,"uvIndex":0,"visibility":10.219,"ozone":334},"hourly":{"summary":"Partly cloudy throughout the day.","icon":"rain","data":[{"time":1560736800,"summary":"Partly Cloudy","icon":"partly-cloudy-day","precipIntensity":0.06,"precipProbability":0.1,"precipType":"rain",

And via the browser I get this;
{"latitude":-41.3456,"longitude":174.8765,"timezone":"Pacific/Auckland","currently":{"time":1560740182,"summary":"Clear","icon":"clear-day","nearestStormDistance":0,"precipIntensity":0.003,"precipProbability":0.13,"precipType":"rain","temperature":53.44,"apparentTemperature":53.44,"dewPoint":35.8,"humidity":0.51,"pressure":1010.93,"windSpeed":2.67,"windGust":11.59,"windBearing":183,"cloudCover":0.11,"uvIndex":0,"visibility":6.329,"ozone":334.4},"hourly":{"summary":"Partly cloudy throughout the day.","icon":"rain","data":[{"time":1560736800,"summary":"Partly Cloudy","icon":"partly-cloudy-day","precipIntensity":0.003,"precipProbability":0.11,"precipType":"rain","temperature":53.93,"apparentTemperature":53.93,"dewPoint":35.54,"humidity":0.5,"pressure":1010.87,"windSpeed":4.92,"windGust":12.79,"windBearing":212,"cloudCover":0.19,"uvIndex":1,"visibility":5.874,"ozone":333.8},{"time":1560740400,"summary":"Clear","icon":"clear-

The epoch times are within a couple of minutes and I was a little liberal with the gps coordinates but even with precise ones it made no difference. The 2 are definitely returning different data

**Comment by HoracioDos** [comment]
Hello.
I also saw this "Dangerously Windy" almost for the whole week, but in our case, cold south east winds are colliding against warm north winds and we've been having quite strong winds and rains for a week. We call this "sudestada". As I didn't used darksky before I didn't know what to expect, We don't call them dangerous but they can be quite strong.

**Comment by 41south** [comment]
Darksky doesn't appear 
![Screen Shot 2019-06-20 at 7 24 48 PM](https://user-images.githubusercontent.com/1725001/59837849-46455280-9391-11e9-8723-9d9921fd520f.png)
happy at all - now it's showing the forecast for yesterday

**Comment by poblabs** [comment]
I went to your site and it looks like it's updated normally now. The last update timestamp in your darksky json is 1561063523 which was 10 minutes ago for me (from this time of update). 

I also don't see "Dangerously Windy" in your darksky json file anymore either. How does it look to you?

**Comment by 41south** [comment]
Well that's kinda weird. I'm currently in Australia and just checked my
site. The Darksky forecast updated on today's date about 15 mins ago (its
the 21st here and at home) but the first forecast date is still yesterday
the 20th.

**Comment by poblabs** [comment]
The updated timestamp makes sense, the forecast date doesn't. The timestamp in the json file shows yesterday for your first forecast date. That data is coming from DarkSky. Hmm

**Comment by poblabs** [comment]
I used my DarkSky key and your GPS coordinates and I am getting the same results. It's showing me yesterday's forecast for your area. This feels broken on DarkSky's end. Did you end up contacting them on the dangerously windy?

**Comment by poblabs** [comment]
I'm wondering if this is a timezone error with how i'm using moment.js now. Hmm. Let me get a virtual machine setup that's in the NZ timezone.

**Comment by 41south** [comment]
I ran out of time before coming overseas to email them, glad you can
reproduce the forecast thing though

**Comment by poblabs** [comment]
I have been trimming some old code out in the 1.1 beta and I wonder if I trimmed too much on the forecast and ended up removing the station timezone awareness. I'll see what I can find

**Comment by poblabs** [comment]
Was this working for you in 1.0? If so I'm lost on what happened. However, I have a fix which is working for me on my NZ machine. 

It seems that moment.js needs the utc offset on all their elements. Why this worked before? Not sure. 

So you will find that your Weather Snapshots was showing yesterday too (Thursday, 20 June 2019), the earthquake data was wrong timezone, the entire records page was wrong timezone too. 

When you're back give the latest updates a try! 

Screenshot (note it's still June 20th where I am, so this shows your timezone info):

![image](https://user-images.githubusercontent.com/3484775/59888678-dfc33300-9396-11e9-813a-e91085d01afa.png)

**Comment by 41south** [comment]
Ah, right, I obviously hadn't looked that far! I can't tell you when it got
strange but I'll do that update either Sunday or Monday. Thanks!

**Comment by poblabs** [comment]
Any update on this one?

**Comment by 41south** [comment]
Just back in NZ last night so will do the update very shortly :)

**Comment by 41south** [comment]
Didn't like that at all - errors below
`Jun 25 13:20:22 raspberrypi weewx[3936]: Belchertown Extension: version 1.1b3
Jun 25 13:20:22 raspberrypi weewx[3936]: reportengine: Caught unrecoverable exception in generator 'weewx.cheetahgenerator.CheetahGenerator'
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****  'group_dust'
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****  Traceback (most recent call last):
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 204, in run
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      obj.start()
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/reportengine.py", line 300, in start
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      self.run()
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 160, in run
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      ngen = self.generate(gen_dict[section_name], self.gen_ts)
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 234, in generate
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      ngen += self.generate(section[subsection], gen_ts)
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 234, in generate
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      ngen += self.generate(section[subsection], gen_ts)
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 322, in generate
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      default_binding)
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 374, in _getSearchList
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      searchList += obj.get_extension_list(timespan, db_lookup)
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/share/weewx/user/belchertown.py", line 797, in get_extension_list
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      obs_unit = self.generator.converter.group_unit_dict[group]
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****    File "/usr/lib/python2.7/dist-packages/configobj.py", line 554, in __getitem__
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****      val = dict.__getitem__(self, key)
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****  KeyError: 'group_dust'
Jun 25 13:20:22 raspberrypi weewx[3936]:         ****  Generator terminated
Jun 25 13:20:22 raspberrypi weewx[3936]: copygenerator: copied 31 files to /var/www/html/weewx/belchertown`

**Comment by poblabs** [comment]
What's group_dust? Is that from another extension or something?

**Comment by 41south** [comment]
Yeah it's from the FilePile extension that imports AQM data from a txt file. It was working ok 😄 
That error maybe the Air Quality Index in my graphs page? It doesn't have units, it's just a number

**Comment by poblabs** [comment]
Hmm maybe. I'm not familiar with it. Did you update everything for
belchertown or just the files from that commit?

**Comment by 41south** [comment]
Did a full install of the development.zip

**Comment by 41south** [comment]
This is the section out of FilePile.py that attributes the group_units

`import syslog
import weewx
import weewx.units
from weewx.wxengine import StdService
from weeutil.weeutil import to_float

weewx.units.USUnits['group_gas_concentration'] = 'ppm'
weewx.units.MetricUnits['group_gas_concentration'] = 'ppm'
weewx.units.MetricWXUnits['group_gas_concentration'] = 'ppm'
weewx.units.default_unit_format_dict['ppm']  = '%.0f'
weewx.units.default_unit_label_dict['ppm']  = ' ppm'

weewx.units.USUnits['group_dust'] = 'microgramm_per_meter_cubic'
weewx.units.MetricUnits['group_dust'] = 'microgramm_per_meter_cubic'
weewx.units.MetricWXUnits['group_dust'] = 'microgramm_per_meter_cubic'
weewx.units.default_unit_format_dict['microgramm_per_meter_cubic']  = '%.1f'
weewx.units.default_unit_label_dict['microgramm_per_meter_cubic']  = ' \xce\xbcg/m\xc2\xb3'

`

**Comment by 41south** [comment]
Reverted back to 1b3 and no errors. On the positive side, 1b4 did fix the time/date errors with Darksky though

**Comment by poblabs** [comment]
That's not good. Means I removed something maybe I shouldn't have. Can you take your working belchertown.py, zip it and drag and drop it into the reply box here? Then I can take a look at your working one vs. the current one to try and zone in. The line that it's complaining about shouldn't be an issue as far as I know - so there's something else going on.

**Comment by 41south** [comment]
I've realised that the version I've gone back to is way way back 1.0.1b3 so it may have a lot of changes! But it doesn't throw that error so hopefully is useful. Apparently I can't just drop the py file here so this is the complete (working) development zip
[weewx-belchertown-development .zip](https://github.com/poblabs/weewx-belchertown/files/3327475/weewx-belchertown-development.zip)

**Comment by poblabs** [comment]
Just curious, if you go to the released proper 1.0.1 version does it
happen?

**Comment by 41south** [comment]
Full release 1.0.1 has no errors 😄

**Comment by poblabs** [comment]
I see what's going on. I'll attach a test file here soon.

**Comment by 41south** [comment]
Great thanks!

**Comment by poblabs** [comment]
Try this attached version of the skin. It's a full skin upgrade. Let me know

[weewx-belchertown.zip](https://github.com/poblabs/weewx-belchertown/files/3327660/weewx-belchertown.zip)

**Comment by 41south** [comment]
Install errors:

pi@raspberrypi:~ $ sudo wee_extension --install weewx-belchertown.zip
Request to install 'weewx-belchertown.zip'
Extracting from zip archive weewx-belchertown.zip
Traceback (most recent call last):
  File "/usr/bin/wee_extension", line 82, in <module>
    main()
  File "/usr/bin/wee_extension", line 74, in main
    ext.install_extension(options.install)
  File "/usr/share/weewx/weecfg/extension.py", line 114, in install_extension
    raise InstallError("Unable to install from '%s': no common path (the extension archive contains more than a single root directory)" % extension_path)
weecfg.extension.InstallError: Unable to install from 'weewx-belchertown.zip': no common path (the extension archive contains more than a single root directory)

**Comment by poblabs** [comment]
😃 

Try this one:

[weewx-belchertown.zip](https://github.com/poblabs/weewx-belchertown/files/3327673/weewx-belchertown.zip)

**Comment by 41south** [comment]
Install 🆗 
Manual wee_reports run 🆗 
Waiting for an archive period for final test 😄

**Comment by 41south** [comment]
But ......
![Screen Shot 2019-06-26 at 12 31 31 PM](https://user-images.githubusercontent.com/1725001/60142533-5edcbf00-980e-11e9-86e9-196263840da1.png)

**Comment by poblabs** [comment]
Wow really? That's an unexpected behavior. Ok let me check something

**Comment by poblabs** [comment]
Actually, your weewx_data.json is empty now. So the wee_reports worked, but not the archive period. 

Did you restart weewx after upgrading the skin?

**Comment by 41south** [comment]
And this too - lots of decimal places 😆 
![Screen Shot 2019-06-26 at 12 32 41 PM](https://user-images.githubusercontent.com/1725001/60142596-8c296d00-980e-11e9-91a2-82294620f148.png)

**Comment by 41south** [comment]
Yes restarted weewx after the install

**Comment by poblabs** [comment]
Yeah the 2 are related. Something's up with your weewx_data.json. Try another wee_reports. If that works, then we'll need to check the syslog during an archive period to see why it gets stuck

**Comment by 41south** [comment]
Yep it's ok after a wee_reports

**Comment by poblabs** [comment]
Ok something's stuck. If you did restart weewx, then check the syslog after an archive period. There should be an error in there. Usually if wee_reports works, but archive does not, it means weewx needs to be restarted - but you've done that.

**Comment by 41south** [comment]
Here's the log, no errors. It's related to mqtt, it's fine until the real-time data kicks in

Jun 26 12:33:42 raspberrypi systemd[1]: Stopping LSB: weewx weather system...
Jun 26 12:33:42 raspberrypi weewx[3653]: engine: Main loop exiting. Shutting engine down.
Jun 26 12:33:42 raspberrypi weewx[3653]: engine: Shutting down StdReport thread
Jun 26 12:33:42 raspberrypi weewx[3653]: engine: Terminating weewx version 3.9.1
Jun 26 12:33:47 raspberrypi weewx[3816]: Stopping weewx weather system: weewx..
Jun 26 12:33:47 raspberrypi systemd[1]: Stopped LSB: weewx weather system.
Jun 26 12:33:47 raspberrypi systemd[1]: Starting LSB: weewx weather system...
Jun 26 12:33:48 raspberrypi weewx[3857]: engine: Initializing weewx version 3.9.1
Jun 26 12:33:48 raspberrypi weewx[3857]: engine: Using Python 2.7.13 (default, Sep 26 2018, 18:42:22) #012[GCC 6.3.0 20170516]
Jun 26 12:33:48 raspberrypi weewx[3857]: engine: Platform Linux-4.14.71-v7+-armv7l-with-debian-9.4
Jun 26 12:33:48 raspberrypi weewx[3857]: engine: Locale is 'en_NZ.UTF-8'
Jun 26 12:33:48 raspberrypi weewx[3857]: engine: pid file is /var/run/weewx.pid
Jun 26 12:33:48 raspberrypi weewx[3846]: Starting weewx weather system: weewx.
Jun 26 12:33:48 raspberrypi systemd[1]: Started LSB: weewx weather system.
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Using configuration file /etc/weewx/weewx.conf
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Loading station type Vantage (weewx.drivers.vantage)
Jun 26 12:33:48 raspberrypi weewx[3862]: filepile: Using /home/pi/AQIData.txt with the 'METRIC' unit system
Jun 26 12:33:48 raspberrypi weewx[3862]: filepile: Label map is {}
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: StdConvert target unit is 0x10
Jun 26 12:33:48 raspberrypi weewx[3862]: wxcalculate: The following values will be calculated: barometer=prefer_hardware, windchill=prefer_hardware, dewpoint=prefer_hardware, appTemp=software, rainRate=prefer_hardware, windrun=prefer_hardware, heatindex=prefer_hardware, maxSolarRad=prefer_hardware, humidex=prefer_hardware, pressure=prefer_hardware, inDewpoint=prefer_hardware, ET=prefer_hardware, altimeter=prefer_hardware, cloudbase=prefer_hardware
Jun 26 12:33:48 raspberrypi weewx[3862]: wxcalculate: The following algorithms will be used for calculations: altimeter=aaNOAA, maxSolarRad=RS
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Archive will use data binding wx_binding
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Record generation will be attempted in 'software'
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Using archive interval of 300 seconds (software record generation)
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Using binding 'wx_binding' to database 'weewx.sdb'
Jun 26 12:33:48 raspberrypi weewx[3862]: manager: Starting backfill of daily summaries
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: StationRegistry: Station will be registered.
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: Wunderground: Posting not enabled.
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: PWSweather: Posting not enabled.
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: CWOP: Data for station ZL2ARL will be posted
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: WOW: Posting not enabled.
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: AWEKAS: Posting not enabled.
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: MQTT: service version is 0.18rc1
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: MQTT: binding to loop
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: MQTT: topic is weather/41south
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: MQTT: desired unit system is METRIC
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: MQTT: data will be uploaded to mqtt://192.168.20.121:1883/
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: Windy: version is 0.4
Jun 26 12:33:48 raspberrypi weewx[3862]: restx: Windy: Data will be uploaded to https://stations.windy.com/pws/update
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Starting up weewx version 3.9.1
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Clock error is -0.69 seconds (positive is fast)
Jun 26 12:33:48 raspberrypi weewx[3862]: engine: Starting main packet loop.
Jun 26 12:35:15 raspberrypi weewx[3862]: manager: Added record 2019-06-26 12:35:00 NZST (1561509300) to database 'weewx.sdb'
Jun 26 12:35:15 raspberrypi weewx[3862]: manager: Added record 2019-06-26 12:35:00 NZST (1561509300) to daily summary in 'weewx.sdb'
Jun 26 12:35:16 raspberrypi weewx[3862]: restx: CWOP: Published record 2019-06-26 12:35:00 NZST (1561509300)
Jun 26 12:35:16 raspberrypi weewx[3862]: restx: Windy: Published record 2019-06-26 12:35:00 NZST (1561509300)
Jun 26 12:35:17 raspberrypi weewx[3862]: restx: StationRegistry: Published record 2019-06-26 12:35:00 NZST (1561509300)
Jun 26 12:35:21 raspberrypi weewx[3862]: cheetahgenerator: Generated 14 files for report StandardReport in 5.11 seconds
Jun 26 12:35:22 raspberrypi weewx[3862]: imagegenerator: Generated 12 images for StandardReport in 0.91 seconds
Jun 26 12:35:22 raspberrypi weewx[3862]: copygenerator: copied 14 files to /var/www/html/weewx
Jun 26 12:35:22 raspberrypi weewx[3862]: Belchertown Extension: version 1.1b3
Jun 26 12:35:32 raspberrypi weewx[3862]: cheetahgenerator: Generated 11 files for report Belchertown in 10.01 seconds
Jun 26 12:35:32 raspberrypi weewx[3862]: copygenerator: copied 31 files to /var/www/html/weewx/belchertown
Jun 26 12:36:30 raspberrypi wee_reports[3935]: cheetahgenerator: Generated 14 files for report StandardReport in 5.06 seconds
Jun 26 12:36:31 raspberrypi wee_reports[3935]: imagegenerator: Generated 12 images for StandardReport in 0.91 seconds
Jun 26 12:36:31 raspberrypi wee_reports[3935]: copygenerator: copied 14 files to /var/www/html/weewx
Jun 26 12:36:31 raspberrypi wee_reports[3935]: Belchertown Extension: version 1.1b3
Jun 26 12:36:41 raspberrypi wee_reports[3935]: cheetahgenerator: Generated 11 files for report Belchertown in 10.11 seconds
Jun 26 12:36:41 raspberrypi wee_reports[3935]: copygenerator: copied 31 files to /var/www/html/weewx/belchertown
Jun 26 12:40:16 raspberrypi weewx[3862]: manager: Added record 2019-06-26 12:40:00 NZST (1561509600) to database 'weewx.sdb'
Jun 26 12:40:16 raspberrypi weewx[3862]: manager: Added record 2019-06-26 12:40:00 NZST (1561509600) to daily summary in 'weewx.sdb'
Jun 26 12:40:17 raspberrypi weewx[3862]: restx: Windy: Published record 2019-06-26 12:40:00 NZST (1561509600)
Jun 26 12:40:19 raspberrypi weewx[3862]: cheetahgenerator: Generated 14 files for report StandardReport in 2.23 seconds
Jun 26 12:40:20 raspberrypi weewx[3862]: imagegenerator: Generated 12 images for StandardReport in 0.86 seconds
Jun 26 12:40:20 raspberrypi weewx[3862]: copygenerator: copied 0 files to /var/www/html/weewx
Jun 26 12:40:23 raspberrypi weewx[3862]: cheetahgenerator: Generated 11 files for report Belchertown in 3.12 seconds
Jun 26 12:40:23 raspberrypi weewx[3862]: copygenerator: copied 2 files to /var/www/html/weewx/belchertown

**Comment by poblabs** [comment]
Ok standby. I've got another idea, may need ~5 min

I can't replicate this since I don't have the `group_dust` observation group

**Comment by poblabs** [comment]
Replace only your belchertown.py with this attached one. Restart weewx. Run a wee_reports (it should work), then let the archive interval happen. 

[belchertown.zip](https://github.com/poblabs/weewx-belchertown/files/3327711/belchertown.zip)

**Comment by 41south** [comment]
Ok, done, just waiting for the archive to run, no problems so far

**Comment by 41south** [comment]
And ...... we're good! 👍  No problems that I can see 🥇

**Comment by poblabs** [comment]
Awesome! 

Summary: I added a new section to weewx_data.json which tells the page how many decimals to round to from your weewx.conf. Offering more consistent control over the data. Your `group_dust` thew everything off, so I added some error handling. The first time I "broke" the loop which broke everything. The second time (just now) I told it to ignore the error and just move on (`continue`). I should have done the continue the first time around - not sure why I did a break. But glad it's working!

I think your dangerously windy is all set too?
